src/handlers.js

```diff
@@ -380,30 +424,54 @@ async function jobHandler(message) {
     await this.createTasks({scopes: graphConfig.scopes, tasks: graphConfig.tasks});
   } catch (e) {
     debug('Creating tasks failed! Leaving comment on Github.');
-    groupState = 'failure';
+    groupState = {
+      status: 'completed',
+      conclusion: 'failure',
+      completed_at: new Date().toISOString(),
+    };
     await this.createExceptionComment({instGithub, organization, repository, sha, error: e});
   } finally {
-    debug(`Trying to create status for ${organization}/${repository}@${sha} (${groupState})`);
+    debug(`Trying to create check runs for ${organization}/${repository}@${sha} (${groupState})`);
     let eventType = message.payload.details['event.type'];
-    let statusContext = `${this.context.cfg.app.statusContext} (${eventType.split('.')[0]})`;
-    let description = groupState === 'pending' ? `TaskGroup: Pending (for ${eventType})` : 'TaskGroup: Exception';
-    await instGithub.repos.createStatus({
-      owner: organization,
-      repo: repository,
-      sha,
-      state: groupState,
-      target_url: INSPECTOR_URL + taskGroupId,
-      description,
-      context: statusContext,
-    });
+
+    await Promise.all(graphConfig.tasks.map(async (task, i) => {
+      const checkRun = await instGithub.checks.create(Object.assign( // TODO: need spread syntax
+        {
+          owner: organization,
+          repo: repository,
+          name: `Task ${i}: ${this.context.cfg.app.statusContext} (${eventType.split('.')[0]})`,
+          head_sha: sha,
+          output: { // TODO: maybe a more helpful output?
```

Comment:
We can probably stick logs from our tasks into this? At least that's my understanding of the api.

Reply:
I wonder if logs would be too large for github?  I was vaguely thinking that as a follow-up we could define a specific artifact that would be reflected into the status, including output and annotations.  Then users could choose what output to include.